crates/lair_keystore_api/src/lair_server.rs

```diff
@@ -95,3 +95,170 @@ use priv_srv::*;
 
 mod priv_api;
 use priv_api::*;
+
+/// Helper for managing a signature_fallback sub-process
+#[derive(Clone)]
+pub(crate) struct FallbackCmd {
+    req: tokio::sync::mpsc::Sender<(
+        LairApiReqSignByPubKey,
+        tokio::sync::oneshot::Sender<LairResult<LairApiResSignByPubKey>>,
+    )>,
+    child: Arc<tokio::process::Child>,
+}
+
+impl FallbackCmd {
+    /// spawn a new sub-process for fallback signature requests
+    pub(crate) async fn new(config: &LairServerConfig) -> LairResult<Self> {
+        let (program, args) = match &config.signature_fallback {
+            LairServerSignatureFallback::Command { program, args } => {
+                (program.clone(), args.clone())
+            }
+            oth => {
+                return Err(format!(
+                    "invalid signature_fallback type: {:?}",
+                    oth,
+                )
+                .into());
+            }
+        };
+
+        let program = dunce::canonicalize(program)?;
+        let args = args.unwrap_or_else(Vec::new);
+
+        // spawn the actual sub-process
+        let mut child = tokio::process::Command::new(program)
+            .args(args)
+            .kill_on_drop(true)
+            .stdin(std::process::Stdio::piped())
+            .stdout(std::process::Stdio::piped())
+            .stderr(std::process::Stdio::null())
```

Comment:
Would this not be better as inherit instead of null as you won't see any errors from the sub process?

Reply:
My thinking was that any errors should preferably be returned as proper error responses to the requests... but you're right, if there's a panic or something that would get lost... will update.

Reply:
done.